consumer/consumererror/consumererror.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/grpc/status"
+)
+
+func New(errs ...error) error {
+	return errors.Join(errs...)
+}
```

Comment:
My primary intent here is that it would be possible to use the `consumererror` package for everything just as a way of making the API a little more obvious. Otherwise you need to utilize two packages (albeit one is a standard library). For example:

```golang
consumererror.New(consumerError.NewPartial([...]), consumererror.NewHTTPStatus([...]), myComponentErr)
```

That said, I'm fine omitting this and just using `errors.Join` directly.

Reply:
All else equal less API seems preferable to me. We also don't have a way to enforce people not using `errors.Join`, so we may as well use that as the default way to go